src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -64,29 +65,65 @@ This hook services the following events:
 # are not present.
 
 # Module imports
-import sys
-import os
-import fcntl
-import stat
-import errno
-import signal
-import subprocess
-import re
-import glob
-import time
-import string
-import platform
-import traceback
-import copy
-import operator
-import pwd
-import fnmatch
-try:
-    import json
-except Exception:
-    import simplejson as json
+#
+# This one is needed to log messages
 import pbs
 
+# if we are not on a Linux system then the hook should always do nothing
+# and accept, and certainly not try Linux-only module imports
+import platform
+if platform.system() != 'Linux':
+    pbs.logmsg(pbs.EVENT_DEBUG,
+               'Cgroup hook not on supported OS '
+               '-- just accepting event')
+    pbs.event().accept()
+
+# we're on Linux, but does the kernel support cgroups?
+rel = list(map(int, (platform.release().split('-')[0].split('.'))))
+pbs.logmsg(pbs.EVENT_DEBUG4,
+           'Cgroup hook: detected Linux kernel version %d.%d.%d' %
+           (rel[0], rel[1], rel[2]))
+supported = False
+if rel[0] > 2:
+    supported = True
+elif rel[0] == 2:
+    if rel[1] > 6:
+        supported = True
+    elif rel[1] == 6:
+        if rel[2] >= 28:
+            supported = True
+if not supported:
+    pbs.logmsg(pbs.EVENT_DEBUG,
+               'Cgroup hook: kernel %s.%s.%s < 2.6.28; not supported'
+               % (rel[0], rel[1], rel[2]))
+    pbs.event.accept()
+else:
+    # Now we know that at least the hook might do something useful
+    # so import other modules (some of them Linux-specific)
+    # Note the else is needed to be PEP8-compliant:  the imports must
+    # not be unindented since they are not top-of-file
```

Comment:
Didn't realize that. Thank you for adding the comment.

Reply:
I only know because pycodestyle complained ;-).